File: nazliyavuz-platform/frontend/nazliyavuz_app/lib/models/message_thread.dart
import 'package:equatable/equatable.dart';
import 'message.dart';
import 'user.dart';

class MessageThread extends Equatable {
  final int id;
  final int chatId;
  final int rootMessageId;
  final String? threadTitle;
  final bool isActive;
  final int messageCount;
  final DateTime? lastActivityAt;
  final DateTime createdAt;
  final DateTime updatedAt;
  final Message? rootMessage;
  final List<Message>? messages;

  const MessageThread({
    required this.id,
    required this.chatId,
    required this.rootMessageId,
    this.threadTitle,
    required this.isActive,
    required this.messageCount,
    this.lastActivityAt,
    required this.createdAt,
    required this.updatedAt,
    this.rootMessage,
    this.messages,
  });

  factory MessageThread.fromJson(Map<String, dynamic> json) {
    return MessageThread(
      id: json['id'] as int,
      chatId: json['chat_id'] as int,
      rootMessageId: json['root_message_id'] as int,
      threadTitle: json['thread_title'] as String?,
      isActive: json['is_active'] as bool? ?? true,
      messageCount: json['message_count'] as int? ?? 0,
      lastActivityAt: json['last_activity_at'] != null 
          ? DateTime.parse(json['last_activity_at']) 
          : null,
      createdAt: DateTime.parse(json['created_at']),
      updatedAt: DateTime.parse(json['updated_at']),
      rootMessage: json['root_message'] != null 
          ? Message.fromJson(json['root_message']) 
          : null,
      messages: json['messages'] != null 
          ? (json['messages'] as List)
              .map((m) => Message.fromJson(m))
              .toList()
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'chat_id': chatId,
      'root_message_id': rootMessageId,
      'thread_title': threadTitle,
      'is_active': isActive,
      'message_count': messageCount,
      'last_activity_at': lastActivityAt?.toIso8601String(),
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
      if (rootMessage != null) 'root_message': rootMessage!.toJson(),
      if (messages != null) 'messages': messages!.map((m) => m.toJson()).toList(),
    };
  }

  @override
  List<Object?> get props => [
        id,
        chatId,
        rootMessageId,
        threadTitle,
        isActive,
        messageCount,
        lastActivityAt,
        createdAt,
        updatedAt,
        rootMessage,
        messages,
      ];
}

File: nazliyavuz-platform/backend/app/Models/MessageThread.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class MessageThread extends Model
{
    use HasFactory;

    protected $fillable = [
        'chat_id',
        'root_message_id',
        'thread_title',
        'is_active',
        'message_count',
        'last_activity_at',
    ];

    protected $casts = [
        'is_active' => 'boolean',
        'last_activity_at' => 'datetime',
    ];

    /**
     * Get the chat that owns the thread.
     */
    public function chat()
    {
        return $this->belongsTo(Chat::class);
    }

    /**
     * Get the root message of the thread.
     */
    public function rootMessage()
    {
        return $this->belongsTo(Message::class, 'root_message_id');
    }

    /**
     * Get all messages in this thread.
     */
    public function messages()
    {
        return $this->hasMany(Message::class, 'thread_id', 'id')
            ->orderBy('created_at');
    }

    /**
     * Get the latest message in the thread.
     */
    public function latestMessage()
    {
        return $this->hasOne(Message::class, 'thread_id', 'id')
            ->latest();
    }

    /**
     * Scope for active threads.
     */
    public function scopeActive($query)
    {
        return $query->where('is_active', true);
    }

    /**
     * Update thread activity.
     */
    public function updateActivity()
    {
        $this->update([
            'last_activity_at' => now(),
            'message_count' => $this->messages()->count(),
        ]);
    }
}
